### ISSUE TITLE
Unexpected warning on "mutable unneeded"

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust#41404]
The following code:

```
pub fn add_old_test(&mut self, mut line: usize, filename: String) {
    let tmp_name = self.generate_name_beginning(&filename);
    let mut number = 0;
    if line > 1 {
        for test in &self.old_tests {
            if test.starts_with(&tmp_name) {
                number += 1;
            }
        }
    }
    if number < line {
        line -= number;
    }
    let name = self.generate_name(line, &filename);
    self.old_tests.push(name);
}
```

Outputs the following warning:

```
error: variable does not need to be mutable
   --> src/librustdoc/html/markdown.rs:495:21
    |
495 |                     mut line: libc::size_t) {
    |                     ^^^^^^^^
    |
    = note: #[deny(unused_mut)] implied by #[deny(warnings)]
```

As far as I can tell, `line` needs to be mutable (not every time though). Is it normal then to get this warning?

**Comment by GuillaumeGomez** [comment]
My bad, I just don't know how to read...
